antarest/study/business/allocation_management.py

```
@@ -0,0 +1,95 @@
+    allocation: List[AllocationField]
+
+
+class AllocationManager:
+    """
+    Provides functionality for getting and updating allocation fields for a given area
+    """
+
+    def __init__(self, storage_service: StudyStorageService) -> None:
+        self.storage_service = storage_service
+
+    def get_field_values(
+        self, study: Study, area_id: str, all_areas
+    ) -> JSONResponse:
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        allocation_data = file_study.tree.get(
+            f"input/hydro/allocation/{area_id}".split("/"), 2
+        )
+
+        areas_ids = [area.id for area in all_areas]
+
+        if not allocation_data:
+            error_msg = f"No allocation data found for area {area_id}"
+            return JSONResponse(content={"error": error_msg}, status_code=404)
+
+        area, values = allocation_data.popitem()
+
+        allocation_fields = AllocationFormFields(
+            allocation=[
+                AllocationField.construct(areaId=area, coefficient=value)
+                for area, value in values.items()
+                if area in areas_ids  # filter invalid areas
+            ]
+        )
+
+        return JSONResponse(
+            content=allocation_fields.dict(), media_type="application/json"
+        )
+
+    def set_field_values(
+        self,
+        study: Study,
+        area_id: str,
+        fields: AllocationFormFields,
+        all_areas: List,
+    ) -> JSONResponse:
+        """
+        Update allocation fields for a given area
+        """
+        areas_ids = [area.id for area in all_areas]
+        allocation_ids = [field.area_id for field in fields.allocation]
+
+        if not set(allocation_ids).issubset(set(areas_ids)):
+            invalid_ids = set(allocation_ids) - set(areas_ids)
+            error_msg = f"Invalid areas: {', '.join(invalid_ids)}"
+            return JSONResponse(content={"error": error_msg}, status_code=422)
```

should raise an exception